src/uwtools/set_config.py

```diff
@@ -93,9 +93,22 @@ def create_config_obj(argv):
 
         config_obj.update_values(user_config_obj)
 
-    if user_args.outfile:
+    if user_args.dry_run:
+        logfile = os.path.join(os.path.dirname(__file__), "set_config.log")
+        log = Logger(level='info',
```

Comment:
Hmm, I'm not sure we want to limit the logger use to only the dry run. I would be ok with either just rendering the output to stdout or initializing the logger at the start